docs/concepts/app-anatomy.md

```diff
@@ -0,0 +1,214 @@
+- Initialize the application's [query routes](./baseapp.md#query-routing) with the [`queriers`](#querier) of each of the application's modules. When a user query comes in, it is routed to the appropriate module using the query routes defined here. 
+- Set the application's [`initChainer`](#initchainer) and mount the stores. 
+- Return the application. 
+
+Note that this function only creates an instance of the app, while the actual state is either carried over from the `~/.appd/data` folder if the node is restarted, or generated from the genesis file if the node is started for the first time. 
+
+You can see an example of application constructor [here](https://github.com/cosmos/sdk-application-tutorial/blob/master/app.go#L46-L128).
+
+### InitChainer
+
+The `initChainer` is a function that initializes the state of the application from a [genesis file](./genesis.md) (i.e. token balances of genesis accounts). It is called when the application received the `InitChain` message from the Tendermint engine, which happens when the node is started at `appBlockHeight == 0` (i.e. on genesis). The application must set the `initChainer` in its constructor via the [`setInitChainer`](https://godoc.org/github.com/cosmos/cosmos-sdk/baseapp#BaseApp.SetInitChainer) method. 
+
+You can see an example of an `initChainer` [here](https://github.com/cosmos/sdk-application-tutorial/blob/master/app.go#L137-L155).
+
+### Register Codec
+
+The `MakeCodec` function is the last important function of the `app.go` file. The goal of this function is to instantiate a codec `cdc` (e.g. [amino](./amino.md)) and calls the `RegisterCodec(*codec.Codec)` method of each module used within the application to register `cdc` to each module. 
+
+In turn, the `RegisterCodec` function of each module register the custom interfaces and type structures of their respective module so that they can be marhsaled and unmarshaled. 
+
+You can see an example of a `MakeCodec` [here](You can see an example of an `initChainer` [here](https://github.com/cosmos/sdk-application-tutorial/blob/master/app.go#L189-L198).).
+
+## Modules
+
+Modules are the heart and soul of an SDK application. They can be considered as state-machines within the state-machine. When a transaction is relayed from the underlying Tendermint engine via the ABCI to the application, it is routed by `baseapp` to the appropriate module in order to be processed. This paradigm enables developers to easily build complex state-machines, as most of the modules they need often already exist. For developers, most of the work involved in building an SDK application revolves around building custom modules required by their application that do not exist, and integrating them with modules that do already exist into one coherent application. In the application directory, the standard practice is to store modules in the `x/` folder. 
```

Comment:
```suggestion
Modules are the heart and soul of an SDK application. They can be considered as state-machines within the state-machine. When a transaction is relayed from the underlying Tendermint engine via ABCI to the application, it is routed by `baseapp` to the appropriate module in order to be processed. This paradigm enables developers to easily build complex state-machines, as most of the modules they need often already exist. For developers, most of the work involved in building an SDK application revolves around building custom modules required by their application that do not exist, and integrating them with modules that do already exist into one coherent application. In the application directory, the standard practice is to store modules in the `x/` folder. 
```

Reply:
So sdk modules are stored in the _the_ `x/` folder but for a custom applications they would be stored in _a_ `x/` folder on the root level of your application... might be good to make this distinction here